### PR TITLE
Add PercentilesBucket aggregation

### PR DESCRIPTION
**Pull request by mkosiedowski**
This adds support for PercentilesBucket - https://github.com/ruflin/Elastica/issues/1805

### CODE REVIEW

File: src/Aggregation/PercentilesBucket.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            throw new InvalidException('Buckets path is required');
+        }
+
+        return parent::toArray();
+    }
+
+    /**
+     * Set the buckets_path for this aggregation.
+     */
+    public function setBucketsPath(string $bucketsPath): PercentilesBucket
```
Comment (deguif) [inline src/Aggregation/PercentilesBucket.php:42]:
```suggestion
    public function setBucketsPath(string $bucketsPath): self
```

File: src/Aggregation/PercentilesBucket.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    {
+        return $this->setParam('buckets_path', $bucketsPath);
+    }
+
+    /**
+     * Set the gap policy for this aggregation.
+     */
+    public function setGapPolicy(string $gapPolicy): PercentilesBucket
```
Comment (deguif) [inline src/Aggregation/PercentilesBucket.php:50]:
```suggestion
    public function setGapPolicy(string $gapPolicy): self
```

File: src/Aggregation/PercentilesBucket.php [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    {
+        return $this->setParam('gap_policy', $gapPolicy);
+    }
+
+    /**
+     * Set the format for this aggregation.
+     */
+    public function setFormat(string $format): PercentilesBucket
```
Comment (deguif) [inline src/Aggregation/PercentilesBucket.php:58]:
```suggestion
    public function setFormat(string $format): self
```

File: src/Aggregation/PercentilesBucket.php [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    {
+        return $this->setParam('format', $format);
+    }
+
+    /**
+     * Set which percents must be returned.
+     *
+     * @param float[] $percents
+     */
+    public function setPercents(array $percents): PercentilesBucket
```
Comment (deguif) [inline src/Aggregation/PercentilesBucket.php:68]:
```suggestion
    public function setPercents(array $percents): self
```

File: src/Aggregation/PercentilesBucket.php [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    {
+        return $this->setParam('percents', $percents);
+    }
+
+    /**
+     * Set keyed flag to return the range as an hash instead of an array of key-value pairs.
+     */
+    public function setKeyed(bool $keyed): PercentilesBucket
```
Comment (deguif) [inline src/Aggregation/PercentilesBucket.php:76]:
```suggestion
    public function setKeyed(bool $keyed): self
```

**Comment by mkosiedowski** [comment]
Thanks for your suggestions @deguif! I've updated the code. I can see that one of the jobs failed on Travis, but it seems that it's not related to changes in the code, but rather to some problem in the test-runner (all other jobs run successfully)

**Comment by ruflin** [comment]
@mkosiedowski Thanks for the contribution. I restarted the failing test. But was not sure if it might be related to a recent change in master and you might have to rebase on master? Lets see if it goes green or not. If not, could you rebase on master and push again?

**Comment by mkosiedowski** [comment]
Thanks @ruflin! It seems that it's all green now :)

**Comment by ruflin** [comment]
Merged! Thanks for the contribution.

**Pull request merged**